lint-staged.config.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+const micromatch = require( 'micromatch' );
+
+/**
+ * Internal dependencies
+ */
+const { plugins } = require( './plugins.json' );
+
+/**
+ * Join and escape filenames for shell.
+ *
+ * @param {Array} files Files to join.
+ *
+ * @return {string} Joined files.
+ */
+const joinFiles = ( files ) => {
+	return files.map( ( file ) => `'${ file }'` ).join( ' ' );
```

Comment:
Oh, I see in https://github.com/lint-staged/lint-staged/pull/875 that an `escape()` function was added. So this should rather be:

```suggestion
	return files.map( escape ).join( ' ' );
```

Along with the `require` above.

Reply:
The above PR was not merged. And the paths that come in the files array are already normalized. I am not sure if we need to rethink this.

Reply:
Oops, I missed that it wasn't merged.

I tried staging 4 files with irregular filenames:

* foo\bar.php
* food bard.php
* food"bard.php
* food'bard.php

When I then tried to commit, I see that they are passed to a command as follows:

```bash
npm run lint:php \
'/home/westonruter/repos/performance/foo/bar.php' \
'/home/westonruter/repos/performance/food bard.php' \
'/home/westonruter/repos/performance/food"bard.php' \
'/home/westonruter/repos/performance/food'bard.php'
```

So interestingly, it normalizes the backslash to a forward slash which makes the file not found. In the second case, space is accounted for by the quotes. In the last example, the single quote is an error because it is not escaped.

Granted, it is highly unlikely for any of these scenarios to be encountered. But to guard against them, additional escaping would be needed since it doesn't seem the normalization in lint-staged is doing any escaping.

Reply:
Not sure how to implement it correctly to support it on both Posix and Win. Even if I add a workaround to handle the above cases, there might be other cases that can break things on different OS.

IMO we should keep it as it is and wait to get it fixed upstream.  Also, if such a file exists and `lint-staged` fails, the user can learn about it and either try to fix the file name or pass a `--no-verify`.

_Found an issue here where it was discussed - https://github.com/nodejs/node/issues/34840._

Reply:
> So interestingly, it normalizes the backslash to a forward slash which makes the file not found.

`lint-staged` does the normalizing here. Should this be reported upstream? But I think the author can rule it out by mentioning it's an irregular file name.

Reply:
Yeah, it's highly irregular. Doubtful anyone will have an issue there.